src/repr/src/adt/datetime.rs

```diff
@@ -96,7 +96,9 @@ impl FromStr for DateTimeUnits {
             "isodow" => Ok(Self::IsoDayOfWeek),
             "isodoy" => Ok(Self::IsoDayOfYear),
             "h" | "hour" | "hours" | "hr" | "hrs" => Ok(Self::Hour),
-            "microsecond" | "microseconds" => Ok(Self::Microseconds),
+            "us" | "usec" | "microsecond" | "microseconds" | "useconds" | "usecs" => {
```

Comment:
I agree, that's a good change. I just modelled it to have the same order as milliseconds down below. So I'll change the order for both of those.

BTW, I've seen other codebases use [burntsushi's fst](https://github.com/burntsushi/fst) when there are so many strings to match since the code generated from a match expression doesn't turn the strings into a trie, but since the statements just get parsed once I don't think there's any benefit from using it.

Reply:
Indeed, in this case it's basically part of the query compilation step so hopefully isn't often in a hot path. We generally like to have benchmarks or other evidence for optimizations that make code meaningfully more complex (even if only under the hood). We should keep it in m ind, though, there might be some places where it would help!